app/frontend/wallet/shelley-wallet.ts

```diff
@@ -71,6 +82,19 @@ const ShelleyWallet = ({config, cryptoProvider}: WalletParams) => {
     )
   }
 
+  function getTokensMetadata(
+    accountInfo: Array<AccountInfo>
+  ): Promise<{[subject: string]: RegisteredTokenMetadata}> {
+    return tokenRegistry.getTokensMetadata([
```

Comment:
the token registry API is not verified to be stable (and we don't know their exact rate limit yet) so I'd rather not risk the app loading to fail if this non-critical piece of information is missing, so I'd add a fall back to catch the error 500 and just return an empty dictionary. The downside would be users wouldn't see their tokens "nicely" but the wallet would still load and they would be able to interact with it.

Reply:
Yes, it's handled here: https://github.com/vacuumlabs/adalite/blob/816389d42b0872f4b8e00fb524f27b68aed0b75b/app/frontend/tokenRegistry/tokenRegistry.ts#L40